src/main/java/org/aksw/simba/lemming/metrics/single/MaxVertexDegreeMetric.java

```diff
@@ -28,5 +34,186 @@ public double apply(ColouredGraph graph) {
             return graph.getGraph().getMaxOutEdgeDegrees();
         }
     }
+    
+    /** Stores the previously computed values in UpdateMetricResult object and checks if degrees is not updated for vertices in candidate set.
+     * @param graph - input graph.
+     * @param graphOperation - boolean value indicating graph operation. ("true" for adding an edge and "false" for removing an edge)
+     * @param triple - Edge on which graph operation is performed.
+     * @param newMetricResult - UpdatableMetricResult object containing the results that should be updated.
+     * @return UpdatableMetricResult object with updated candidate set and metric values.
+     */
+    @Override
+    public UpdatableMetricResult applyUpdatable(ColouredGraph graph, boolean graphOperation, TripleBaseSingleID triple, UpdatableMetricResult newMetricResult) {
+    	MaxVertexDegreeMetricResult metricResultTempObj = (MaxVertexDegreeMetricResult) newMetricResult;
+    	if(graphOperation) {
+    		metricResultTempObj.setCandidatesMetricAddAnEdge(metricResultTempObj.getmMapCandidatesMetricTemp(),metricResultTempObj.getmMapCandidatesMetricValuesTemp());
+    		metricResultTempObj.verifyCandidates(GRAPHOPERATION.AddAnEdgeIndegree, triple);
+    		metricResultTempObj.verifyCandidates(GRAPHOPERATION.AddAnEdgeOutdegree, triple);
+    	}else {
+    		metricResultTempObj.setCandidatesMetricRemoveAnEdge(metricResultTempObj.getmMapCandidatesMetricTemp(),metricResultTempObj.getmMapCandidatesMetricValuesTemp());
+    		metricResultTempObj.verifyCandidates(GRAPHOPERATION.RemoveAnEdgeIndegree, triple);
+    		metricResultTempObj.verifyCandidates(GRAPHOPERATION.RemoveAnEdgeOutdegree, triple);
```

Comment:
I do not understand why all the functionality is moved into the result object. The metric functionality should be implemented in the metric class. The result object should only contain the result and (maybe) some data that is necessary to create an updated result later on.

Reply:
I have included the implementation of the verifyCandidates method in MaxVertexDegreeMetric class.